packages/wallet/backend/src/transaction/service.ts

```diff
@@ -50,4 +60,36 @@ export class TransactionService implements ITransactionService {
       this.deps.logger.error(`Update transaction error:`, e)
     }
   }
+
+  async listAll({
+    userId,
+    paginationParams: { page, pageSize },
+    filterParams,
+    orderByDate
+  }: ListAllTransactionsInput): Promise<Page<Transaction>> {
+    const filterParamsWithTableNames = prefixSomeObjectKeys(
+      filterParams,
+      ['paymentPointerId', 'assetCode', 'type', 'status', 'accountId'],
+      'transactions.'
+    )
+
+    const transactions = await Transaction.query()
+      .select(
+        'transactions.*',
+        'paymentPointer.url as paymentPointerUrl',
+        'paymentPointer:account.name as accountName'
+      )
+      .fullOuterJoinRelated('[paymentPointer.[account.user], account.user]')
+      .where(function () {
+        this.where('paymentPointer:account:user.id', userId).orWhere(
```

Comment:
should be simplified after adding accountId on every transaction

Reply:
I do not understand how this can simplified at the moment. 🤔 Can you elaborate?

Reply:
as accountId it will be always present only the second check can be kept here and remove the one with `paymentPointer:account:user.id`